app/main.py

```diff
@@ -1,8 +1,35 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.wife = None
+        self.husband = None
+
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    persons = [Person(data["name"], data["age"]) for data in people]
+
+    for data in people:
+        person = Person.people[data["name"]]
+        spouse_name = data.get("wife") or data.get("husband")
+
+        if spouse_name is not None and spouse_name in Person.people:
+            spouse = Person.people[spouse_name]
+            if "wife" in data and spouse_name:
```

Comment:
The condition `if "wife" in data and spouse_name:` is redundant because `spouse_name` is already checked in the previous condition. You can simplify this to `if "wife" in data:`.